### PR TITLE
Reusable team member view(1)

### LINKED ISSUE #16: 🎨 Create reusable team member view (Single team member view) - When clicked
#  Create reusable team member view (Single team member view) - When clicked (Responsive)
---

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user I want a reusable section that shows a team member and some brief info about them

### Description

As seen on the designs (link below), this section contains the following:

```
1. An Image on the left
2. The name and role of the team member
3. Some paragraphs describing the team member
4. A floating button on the image that links to (LinkedIn)
```

The team section needs to appear as shown below.

Screenshot 2023-02-02 at 19 17 45

When the cancel button on the top right corner is clicked the section goes back to view on #41.

Your assignment is to implement the component to appear as above.

Please take note of all the design features 

#### Link to full design:
```
https://xd.adobe.com/view/7b72572b-b40a-4618-bfa6-5500ab28f13b-2e21/screen/63e63506-4ea3-4d02-af28-f7cf148b47d9

mobile 

https://xd.adobe.com/view/7b72572b-b40a-4618-bfa6-5500ab28f13b-2e21/screen/0efb593f-dc28-4e6d-86f3-098699455ad4
```
### Technical Analysis

This component will be created in  libs/features/components/banners

The component will be called elewa-group-team-member

### Acceptance Criteria

1. When rendered the component is pixel perfect compared to the design above.
6. The component is clean and does not have more than 50 lines
7. The component SCSS uses CSS variables for colour, font, ... Connected to the main style.scss
8. The code is isolated to the libs/page/home package, unless a justified reason is specified.
9. The component is aligned using CSS flex
10. The design is responsive for screen, tablet and phone
11. The image renders fast and has multiple srcsets

### PR DESCRIPTION
**Pull request by 23po**
# Description

Created a Reusable Single team member feature. When a user clicks on a specific member of the team from the team member carousel, the single page view of that team member should open. 

The change does not require any new/special dependencies

Fixes  #16 

To achieve this, we needed to:

```
- The change includes a new banners-features-components module 
- A new Elewa-Group-Single-Page-View Component and dependent files (scss, html, typescript files) and configuration
- The Module is integrated with the App module, and set up to display the Single Page View Component
- Finally styling and logic for the
```

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Initial commits of package.lock.json were reset
- [x] Three conflicts to be aware of based on generated modules 

# Screenshots
Desktop View
mobile responsive view

# How Has This Been Tested?
The change works on localhost as illustrated by the screenshots above

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: apps/elewa-group-website/src/_variables.scss [hunk 1]
```diff
@@ -45,4 +45,6 @@
   --elewa-group-website-font-size-description: 13px;
   --elewa-group-website-font-size-description-medium: 12px;
 
+  //reusable team member view
+
 }
```
Comment (IanOdhiambo9) [inline apps/elewa-group-website/src/_variables.scss:50]: This file did not need to change

File: apps/elewa-group-website/src/app/app.component.html [hunk 2]
```diff
@@ -1 +1,5 @@
-<elewa-group-home-page></elewa-group-home-page>
\ No newline at end of file
+<!-- <elewa-group-home-page></elewa-group-home-page>  -->
+<!-- 
+<elewa-group-card></elewa-group-card> -->
+
+<elewa-group-elewa-group-team-member></elewa-group-elewa-group-team-member>
```
Comment (IanOdhiambo9) [inline apps/elewa-group-website/src/app/app.component.html:5]: The tag name of this component is wrong,
Should be -> elewa-group-team-member not elewa-group-elewa-group-team-member

File: libs/features/components/banners/src/lib/elewa-group-team-member/elewa-group-team-member.component.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'elewa-group-elewa-group-team-member',
```
Comment (IanOdhiambo9) [inline libs/features/components/banners/src/lib/elewa-group-team-member/elewa-group-team-member.component.ts:4]: Wrong name on the selector

File: libs/features/components/banners/src/lib/elewa-group-team-member/elewa-group-team-member.component.html [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+               <button class="close-button" (click)="goBackToMultiUserView()">
+                <i class="fa-solid fa-xmark"></i>
+               </button>
+               
+               
+              </button>
+            </div>
+        </div>
+      </ng-container>
+    <div class="banner-controls">
```
Comment (IanOdhiambo9) [inline libs/features/components/banners/src/lib/elewa-group-team-member/elewa-group-team-member.component.html:25]: This entire component should be moved into a components folder inside the lib components/elewa-group-team-member

**Review by IanOdhiambo9 — CHANGES_REQUESTED** [review]
You pushed your package-lock.json
This PR cannot be merged

**Comment by IanOdhiambo9** [comment]
Please resolve conflicts on PR

File: apps/elewa-group-website/src/app/app.component.html (hunk 2 above)
Comment (23po) [reply apps/elewa-group-website/src/app/app.component.html]: fixed to elewa-group-team-member

File: apps/elewa-group-website/src/_variables.scss (hunk 1 above)
Comment (23po) [reply apps/elewa-group-website/src/_variables.scss:50]: this has been resolved using git rebase -i <hash>

File: libs/features/components/banners/src/lib/elewa-group-team-member/elewa-group-team-member.component.ts (hunk 3 above)
Comment (23po) [reply libs/features/components/banners/src/lib/elewa-group-team-member/elewa-group-team-member.component.ts]: fixed

File: apps/elewa-group-website/src/_variables.scss (hunk 1 above)
Comment (23po) [reply apps/elewa-group-website/src/_variables.scss]: Sending new PR to solve this

File: apps/elewa-group-website/src/app/app.module.ts [hunk 5]
```diff
@@ -9,13 +9,16 @@ import { HomePageModule } from '@elewa-group/pages/elewa/home';
...
+
 @NgModule({
   declarations: [AppComponent],
   imports: [
     BrowserModule, 
     ScullyLibModule,
     HomePageModule,
-    CardsModule
+    CardsModule,
+    FeaturesComponentsBannersModule
```
Comment (IanOdhiambo9) [inline apps/elewa-group-website/src/app/app.module.ts:21]: Rename this module to just BannersModule

File: libs/features/components/banners/src/lib/components/elewa-group-team-member/elewa-group-team-member.component.html [hunk 6]
```diff
@@ -0,0 +1 @@
+<p>elewa-group-team-member works!</p>
```
Comment (IanOdhiambo9) [inline libs/features/components/banners/src/lib/components/elewa-group-team-member/elewa-group-team-member.component.html:1]: There's not template code on this file??

File: libs/features/components/banners/src/lib/components/elewa-group-team-member/elewa-group-team-member.component.scss [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+    position: absolute;
+    bottom: 90px;
+    right: 0px;
+    display: flex;
+    align-items: center;
+    margin-left: 5%;
+    margin-right: 5%;
+    padding: 0 40px;  
+  }
+
```
Comment (IanOdhiambo9) [inline libs/features/components/banners/src/lib/components/elewa-group-team-member/elewa-group-team-member.component.scss:88]: The navigation controls need to remain on the right side even on mobile view, the scrollbar at the bottom is also too thick on mobile. Make it a bit thinner.

File: libs/features/components/banners/src/lib/components/elewa-group-team-member/elewa-group-team-member.component.ts [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  }
+  
+  get progressPercentage() {
+    return (this.currentBannerIndex + 1) / this.banners.length * 100;
+  }
+  
+  goBackToMultiUserView() {
+
+    console.log("clicked")
+    // code to change state back to multi-user view
```
Comment (IanOdhiambo9) [inline libs/features/components/banners/src/lib/components/elewa-group-team-member/elewa-group-team-member.component.ts:44]: remove dead code

File: libs/features/components/banners/src/lib/features-components-banners.module.ts [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { ElewaGroupTeamMemberComponent } from './components/elewa-group-team-member/elewa-group-team-member.component';
+
+@NgModule({
+  imports: [CommonModule],
+  declarations: [ElewaGroupTeamMemberComponent],
+  exports: [ElewaGroupTeamMemberComponent]
+})
+export class FeaturesComponentsBannersModule {}
```
Comment (IanOdhiambo9) [inline libs/features/components/banners/src/lib/features-components-banners.module.ts:10]: Rename this to BannersModule

**Review by IanOdhiambo9 — CHANGES_REQUESTED** [review]
Add a short video of the carousel in action on both mobile and web.

File: libs/features/components/banners/src/lib/components/elewa-group-team-member/elewa-group-team-member.component.html (hunk 6 above)
Comment (23po) [reply libs/features/components/banners/src/lib/components/elewa-group-team-member/elewa-group-team-member.component.html:1]: Hi, I did a second PR, which has the most updated code: https://github.com/italanta/elewa-group/pull/132

File: apps/elewa-group-website/src/app/app.module.ts (hunk 5 above)
Comment (23po) [reply apps/elewa-group-website/src/app/app.module.ts:21]: fixed --> https://github.com/italanta/elewa-group/pull/132

File: libs/features/components/banners/src/lib/components/elewa-group-team-member/elewa-group-team-member.component.scss (hunk 7 above)
Comment (23po) [reply libs/features/components/banners/src/lib/components/elewa-group-team-member/elewa-group-team-member.component.scss:88]: fixed --> https://github.com/italanta/elewa-group/pull/132

File: libs/features/components/banners/src/lib/components/elewa-group-team-member/elewa-group-team-member.component.ts (hunk 8 above)
Comment (23po) [reply libs/features/components/banners/src/lib/components/elewa-group-team-member/elewa-group-team-member.component.ts:44]: fixed --> https://github.com/italanta/elewa-group/pull/132

File: libs/features/components/banners/src/lib/features-components-banners.module.ts (hunk 9 above)
Comment (23po) [reply libs/features/components/banners/src/lib/features-components-banners.module.ts:10]: fixed here https://github.com/italanta/elewa-group/pull/132

**Comment by 23po** [comment]
@IanOdhiambo9 requested changes have all been addressed here --> https://github.com/italanta/elewa-group/pull/132

**Pull request closed without merging**